packages/schema/src/model.ts

```diff
@@ -16,23 +16,71 @@ import {
   IJupyterCadModel,
   ISelection,
   IUserData,
-  Pointer
+  Pointer,
+  IJCadSettings
 } from './interfaces';
 import jcadSchema from './schema/jcad.json';
 import { Contents } from '@jupyterlab/services';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+
+const SETTINGS_ID = '@jupytercad/jupytercad-core:jupytercad-settings';
 
 export class JupyterCadModel implements IJupyterCadModel {
   constructor(options: JupyterCadModel.IOptions) {
-    const { annotationModel, sharedModel } = options;
+    const { annotationModel, sharedModel, settingRegistry } = options;
     if (sharedModel) {
       this._sharedModel = sharedModel;
     } else {
       this._sharedModel = this.createSharedModel();
     }
     this._connectSignal();
     this.annotationModel = annotationModel;
+    this.settingRegistry = settingRegistry;
     this._copiedObject = null;
     this._pathChanged = new Signal<JupyterCadModel, string>(this);
+    this._settingsReady = this.initSettings();
+  }
+
+  /**
+   * Initialize custom settings for JupyterLab.
+   */
+  async initSettings(): Promise<void> {
+    if (this.settingRegistry) {
+      this._settings = await this.settingRegistry.load(SETTINGS_ID);
+
+      this._updateLocalSettings();
+
+      this._settings.changed.connect(() => {
+        this._updateLocalSettings();
+      });
+    }
+  }
+
+  private _updateLocalSettings(): void {
+    const composite = this._settings.composite;
+
+    this._jcadSettings = {
+      showAxesHelper: (composite.showAxesHelper as boolean) ?? false,
+      cameraType:
+        (composite.cameraType as 'Perspective' | 'Orthographic') ??
+        'Perspective'
+    };
+  }
+
+  jcadSettings(): IJCadSettings {
+    return this._jcadSettings;
+  }
+
+  get settingsReady(): Promise<void> {
+    return this._settingsReady;
+  }
+
+  /**
+   * Return stored settings.
+   */
+  async getSettings(): Promise<ISettingRegistry.ISettings> {
+    await this.initSettings();
```

Comment:
you re-init the setting every time someone reads the setting?

Reply:
Ahh, i too feel it's wrong. I've done this in [this commit](https://github.com/jupytercad/JupyterCAD/pull/725/commits/ee3ac148de744f7aeacb5abcb7175f036e10f790) while trying to achieve something